syntax = "proto3";

package proto;

enum StorageErrorReason {
  STORAGE_ERROR_REASON_UNKNOWN = 0;
  STORAGE_LIFECYCLE_CHANGE_NOT_ALLOWED = 1;
  STORAGE_INVALID_STATE = 2;
  STORAGE_UNIT_OF_WORK_NOT_SEEDED = 3;
  STORAGE_STATS_COLLECTION_FAILED = 4;
  STORAGE_DEADLINE_EXCEEDED = 5;
  STORAGE_SETTING_F1_SNAPSHOT_TIMESTAMP_IS_DISALLOWED = 6;
  STORAGE_PAGINATED_DISALLOWED_IN_CHILD_REQUEST = 7;
  STORAGE_STALE_LOCK_TIMESTAMP = 11;
  STORAGE_DATA_CONSTRAINT_VIOLATION = 12;
  STORAGE_LEGACY_ENTITY_NOT_FOUND = 13;
  STORAGE_BATCH_READ_WRITE_MIX_NOT_ALLOWED = 14;
  STORAGE_BATCH_INCONSISTENT_CHILD_PAGINATION_TIMESTAMPS = 15;
  STORAGE_BATCH_PARENT_UNIT_OF_WORK_INVALID = 16;
  STORAGE_PAGINATED_DISALLOWED_WITHOUT_SNAPSHOT_TIMESTAMP = 17;
}
